server/graphql/v2/mutation/VirtualCardMutations.ts

```diff
@@ -203,10 +206,26 @@ const virtualCardMutations = {
         UserId = user.id;
       }
 
-      return await virtualCard.update({
-        privateData: { ...(virtualCard.get('privateData') as Object), cvv, expireDate },
+      const updateAttributes = {
+        name: args.name,
         UserId,
-      });
+      };
+
+      if (args.monthlyLimit && virtualCard.spendingLimitInterval === 'MONTHLY' && virtualCard.provider === 'STRIPE') {
+        const monthlyLimitInCents = args.monthlyLimit.valueInCents;
+
+        if (monthlyLimitInCents > 100000) {
+          throw new BadRequest(`Monthly limit should not exceed 1000 ${virtualCard.currency}`, undefined, {
+            monthlyLimit: `Monthly limit should not exceed 1000 ${virtualCard.currency}`,
+          });
+        }
+
+        updateAttributes['spendingLimitAmount'] = monthlyLimitInCents;
+
+        await stripe.updateVirtualCard(virtualCard, monthlyLimitInCents);
```

Comment:
Not a requested change.

It's a bit strange to me to have `updateVirtualCard` and have `monthlyLimitInCents` as single argument. I would have passed it within an object `await stripe.updateVirtualCard(virtualCard, {monthlyLimit});`